spec/promoted/ruby/client_spec.rb

```diff
@@ -151,15 +150,85 @@
   end
 
   context "deliver" do
+    before(:example) do
+      @input = Marshal.load(Marshal.dump(SAMPLE_INPUT_CAMEL))
+    end
+
     it "delivers in a good case" do
       client = described_class.new
-      full_insertion = deliver_input[:fullInsertion]
+      full_insertion = @input[:fullInsertion]
       expect(client).to receive(:send_request).and_return({
         :insertion => full_insertion
       })
-      deliver_resp = client.deliver deliver_input
+      deliver_resp = client.deliver @input
       expect(deliver_resp).not_to be nil
+      expect(deliver_resp.key?(:insertion)).to be true
+
+      # No log request generated since there's no experiment and we delivered the request.
+      expect(deliver_resp[:log_request]).to be nil
     end
+
+    it "can custom compact insertions" do
+      @input[:to_compact_delivery_insertion_func] = described_class.copy_and_remove_properties
+
+      client = described_class.new
+      full_insertion = @input[:fullInsertion]
+      delivery_req = nil
+      allow(client).to receive(:send_request) { |value|
+        delivery_req = value
+        { :insertion => full_insertion }
+      }
+
+      deliver_resp = client.deliver @input
+
+      delivery_req[:insertion].each do |insertion|
+        expect(insertion.key?(:properties)).to be false
+      end
+
+      expect(deliver_resp).not_to be nil
+      expect(deliver_resp.key?(:insertion)).to be true
+
+      # No log request generated since there's no experiment and we delivered the request.
+      expect(deliver_resp[:log_request]).to be nil
+    end
+  end
+
+  context "cohorts" do
+    before(:example) do
+      @input = Marshal.load(Marshal.dump(SAMPLE_INPUT_CAMEL))
+      @input["experiment"] = {
+        "cohortId" => "HOLD_OUT",
+        "arm" => "CHANGE ME"
+      }
+    end
+
+    it "does not deliver for control arm" do
+      client = described_class.new
+      @input["experiment"]["arm"] = Promoted::Ruby::Client::COHORT_ARM['CONTROL']
+      expect(client).not_to receive(:send_request)
+      deliver_resp = client.deliver @input
+      expect(deliver_resp).not_to be nil
+      expect(deliver_resp[:log_request].key?(:insertion)).to be true
+      expect(deliver_resp[:log_request].key?(:request)).to be true
+      expect(deliver_resp[:log_request].key?(:cohort_membership)).to be true
```

Comment:
thoughts on asserting the exact values?  Or not worth not worth it because it's already asserted somewhere else?

Reply:
There are a lot more things that I'd like to add expectations for, first and foremost being the deliver _request_, which I just figured out how to capture from the mock last night. Going to work on those incrementally as I continue to test and finish this SDK.